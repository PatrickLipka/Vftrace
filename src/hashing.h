#ifndef HASHING_H
#define HASHING_H

#include "stack_types.h"

uint64_t vftr_jenkins_murmur_64_hash(size_t length, const uint8_t* key);

void vftr_compute_stack_hashes(int nstacks, stack_t *stacks);

#endif
