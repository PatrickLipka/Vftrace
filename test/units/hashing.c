#include <stdio.h>

#include <string.h>

#include "hashing.h"

int main(int argc, char **argv) {
   (void) argc;
   (void) argv;
   char *string0 = "";
   char *string1 = "A";
   char *string2 = "When on board H.M.S. Beagle, as naturalist, ";
   char *string3 = "I was much struck with certain facts ";
   char *string4 = "in the distribution of the organic beings";
   char *loremipsum =
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, "
      "sed do eiusmod tempor incididunt ut labore et dolore "
      "magna aliqua. Ut enim ad minim veniam, quis nostrud "
      "exercitation ullamco laboris nisi ut aliquip ex ea commodo "
      "consequat. Duis aute irure dolor in reprehenderit in "
      "voluptate velit esse cillum dolore eu fugiat nulla pariatur. "
      "Excepteur sint occaecat cupidatat non proident, sunt in "
      "culpa qui officia deserunt mollit anim id est laborum.";

   printf("%016lx: %s\n",
          vftr_jenkins_murmur_64_hash(strlen(string0), (uint8_t*)string0),
          string0);
   printf("%016lx: %s\n",
          vftr_jenkins_murmur_64_hash(strlen(string1), (uint8_t*)string1),
          string1);
   printf("%016lx: %s\n",
          vftr_jenkins_murmur_64_hash(strlen(string2), (uint8_t*)string2),
          string2);
   printf("%016lx: %s\n",
          vftr_jenkins_murmur_64_hash(strlen(string3), (uint8_t*)string3),
          string3);
   printf("%016lx: %s\n",
          vftr_jenkins_murmur_64_hash(strlen(string4), (uint8_t*)string4),
          string4);
   printf("%016lx: %s\n",
          vftr_jenkins_murmur_64_hash(strlen(loremipsum), (uint8_t*)loremipsum),
          loremipsum);
   return 0;
}
