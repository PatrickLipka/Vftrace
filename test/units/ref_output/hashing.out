000000006e9984af: 
820103f007446fd2: A
cef122340d15b6ac: When on board H.M.S. Beagle, as naturalist, 
f2cee8d774020de0: I was much struck with certain facts 
cbc15404b81162b5: in the distribution of the organic beings
5cba1d5610139dae: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
